api/BUILD

```diff
@@ -10,12 +10,35 @@ bool_flag(
     build_setting_default = False,
 )
 
+CPP_STDLIBS = [
+    "none",
+    "best",
+    "2014",
+    "2017",
+    "2020",
+    "2023",
+]
+
+string_flag(
+    name = "with_cxx_stdlib",
+    values = CPP_STDLIBS,
+    build_setting_default = "best",
+)
+
 cc_library(
     name = "api",
     hdrs = glob(["include/**/*.h"]),
     defines = select({
         ":with_external_abseil": ["HAVE_ABSEIL"],
         "//conditions:default": [],
+    }) + select({
+        ":set_cxx_stdlib_none": [],
+        ":set_cxx_stdlib_best": ["OPENTELEMETRY_STL_VERSION=(__cplusplus/100)"],
```

Comment:
Not a bazel expert, hope you have validated that this will set `OPENTELEMETRY_STL_VERSION` to the correct STL version after divison, and not just the string `(__cplusplus/100)`.

Reply:
The bit in the quotes is passed verbatim to the build command line. (I think via exec without shell processing?) It's the C-processor that evaluates that math to the version and then does any needed comparison.

Tl;dr; passing "just the string" is what it's _supposed_ to do.

https://gcc.godbolt.org/z/ofnbf8joY  << A demo showing that work (try playing around with the `-std=c++11` command line flag).

Reply:
Thanks for the clarification. That makes sense.